integration_test/scenario_flow_test.go

```diff
@@ -88,27 +68,35 @@ func setupApiCreds(ctx context.Context, t *testing.T, usecasesWithCreds usecases
 		Description:    "Test API key",
 		Role:           models.API_CLIENT,
 	})
-	assert.NoError(t, err, "Could not create api key")
+	if err != nil {
+		assert.FailNow(t, "Could not create api key", err)
```

Comment:
If this fails, nothing else after this will work in the test suite.
With assert.NoError, the test will continue to run, generating tons of noise and making it hard to find where the initial error was.